be/src/exec/except_node.cpp

```diff
@@ -56,4 +40,157 @@ Status ExceptNode::init(const TPlanNode& tnode, RuntimeState* state) {
+        _hash_tbl_iterator = _hash_tbl->begin();
+        return Status::OK();
+    }
+
+    for (int i = 1; i < _children.size(); ++i) {
+        // probe
+        _probe_batch.reset(new RowBatch(child(i)->row_desc(), state->batch_size(), mem_tracker()));
+        RETURN_IF_ERROR(child(i)->open(state));
+        eos = false;
+        while (!eos) {
+            RETURN_IF_CANCELLED(state);
+            RETURN_IF_ERROR(child(i)->get_next(state, _probe_batch.get(), &eos));
+            RETURN_IF_LIMIT_EXCEEDED(state, " Except , while probing the hash table.");
+            for (int j = 0; j < _probe_batch->num_rows(); ++j) {
+                _hash_tbl_iterator = _hash_tbl->find(_probe_batch->get_row(j));
+                if (_hash_tbl_iterator != _hash_tbl->end()) {
+                    _hash_tbl_iterator.set_matched();
+                }
+            }
+            _probe_batch->reset();
+        }
+        if (i != _children.size() - 1) {
+            // if this hash table is probed rebuild it
+            std::unique_ptr<HashTable> temp_tbl(
+                    new HashTable(_child_expr_lists[0], _child_expr_lists[i], _build_tuple_size,
+                                  true, _find_nulls, id(), mem_tracker(), 1024));
+            _hash_tbl_iterator = _hash_tbl->begin();
+            uint32_t previous_hash = -1;
+            while (_hash_tbl_iterator.has_next()) {
+                if (previous_hash != _hash_tbl_iterator.get_hash()) {
+                    previous_hash = _hash_tbl_iterator.get_hash();
+                    if (!_hash_tbl_iterator.matched()) {
+                        temp_tbl->insert(_hash_tbl_iterator.get_row());
+                    }
+                }
+                _hash_tbl_iterator.next<false>();
+            }
+            _hash_tbl.swap(temp_tbl);
+            temp_tbl->close();
+        }
+        // if a table is empty, the result must be empty
+        if (_hash_tbl->size() == 0) {
+            break;
+        }
+    }
+    _hash_tbl_iterator = _hash_tbl->begin();
+    return Status::OK();
+}
+
+Status ExceptNode::get_next(RuntimeState* state, RowBatch* out_batch, bool* eos) {
+    RETURN_IF_ERROR(exec_debug_action(TExecNodePhase::GETNEXT));
+    RETURN_IF_CANCELLED(state);
+    SCOPED_TIMER(_runtime_profile->total_time_counter());
+    *eos = true;
+    if (reached_limit()) {
+        return Status::OK();
+    }
+    uint32_t previous_hash = -1;
+    while (_hash_tbl_iterator.has_next()) {
+        if (previous_hash != _hash_tbl_iterator.get_hash()) {
+            previous_hash = _hash_tbl_iterator.get_hash();
+            if (!_hash_tbl_iterator.matched()) {
+                int row_idx = out_batch->add_row();
+                TupleRow* out_row = out_batch->get_row(row_idx);
+                uint8_t* out_ptr = reinterpret_cast<uint8_t*>(out_row);
+                memcpy(out_ptr, _hash_tbl_iterator.get_row(), _build_tuple_row_size);
+                out_batch->commit_last_row();
+            }
+        }
```

Comment:
If the two hash values are equal, it does not mean that the two values are equal. You need to compare the original values.
There may be a problem with the else's logic